keras_cv/layers/spatial_pyramid.py

```diff
@@ -157,47 +143,23 @@ def build(self, input_shape):
     def call(self, inputs, training=None):
         """Calls the Atrous Spatial Pyramid Pooling layer on an input.
 
-        The input the of the layer will be a dict of {`level`, `tf.Tensor`}, and layer
-        will pick the actual input based on the `level` from its init args.
-        The output of the layer will be a dict of {`level`, `tf.Tensor`} with only one level.
-
         Args:
-          inputs: A `dict` of `tf.Tensor` where
-            - key: A `int` of the level of the multilevel feature maps.
-            - values: A `tf.Tensor` of shape [batch, height_l, width_l,
-              filter_size].
+          inputs: A `tf.Tensor` of shape [batch, height, width, channels]
```

Comment:
so you would like the backbone network to directly produce one tensor at the specific feature level. This somehow impose a more strong coupling between the backbone network and the ASPP layer, and user might not easily swap the components.

Reply:
Yeah, because users have to carefully choose the dilation rate for each level.
I think in the future it might be better if we allow creating multiple ASPP layers, and each take different level and dilation rates, but all accept dictionary of inputs -- but I'd probably figure out what's the best API once I start adding DeepLabV3+. For now, it might be better to do the minimal?

Reply:
(The more I think, the more I feel like even in DeepLabV3+, we need something like this:
```
level_to_dilation = {2: [4, 6, 8], 4: [2, 4, 6]}
level_to_aspp = {key: SpatialPyramidPooling(key, value) for key, value in level_to_dilation.items()}
backbone_outputs = backbone(inputs).get(level_to_dilation.keys())
...
```
So passing a dict of inputs (2, 3, 4, 5) and getting a dict of outputs (2) seems weird and confusing to users, instead we accept single tensor and return single tensor seems more straightforward.